tools/server/server-models.cpp

```diff
@@ -811,6 +811,7 @@ void server_models_routes::init_routes() {
         if (name.empty()) {
             // main instance
             auto res = std::make_unique<server_http_res>();
+            json webui_settings = params.webui_config_json.empty() ? json::object() : json::parse(params.webui_config_json);
```

Comment:
calling `json::parse` here will potentially impact the performance if this endpoint is repeatedly called by the client (which is the case in practice)

I would recommend to parse this is `server_context::webui_config` inside `server_context::init()` and reuse it later